File: lecshop_base_modules/lecshop_service/src/main/java/com/lecshop/admin/profitset/bean/ProfitSet.java
package com.lecshop.admin.profitset.bean;

import lombok.Data;

import java.math.BigDecimal;

/**
 * 律师收益设置实体类
 *
 * @author sunluyang on 2017/7/11.
 */
@Data
public class ProfitSet {
    /**
     * 主键id
     */
    private long id;
    /**
     * 每月多少号
     */
    private int withdrawTime;
    /**
     * 电话咨询一次多少钱
     */
    private BigDecimal telConMoney;
    /**
     * 电话咨询一次 律师的佣金设置
     */
    private BigDecimal telConCommission;
    /**
     * 预约会面多少钱 （指定律师）
     */
    private BigDecimal meetMoney;
    /**
     * 预约会面 律师的佣金(指定律师)
     */
    private BigDecimal meetCommission;
    /**
     * 预约会面一次多少钱（不指定律师）
     */
    private BigDecimal meetAllMoney;
    /**
     * 预约会面的佣金 （不指定律师）
     */
    private BigDecimal meetAllCommisson;

    /**
     * 修改文书多少钱
     */
    private BigDecimal modifyDocMoney;
    /**
     * 修改文书的佣金
     */
    private BigDecimal modifyDocCommission;
    /**
     * 草拟文书多少钱
     */
    private BigDecimal draftDocMoney;
    /**
     * 草拟文书佣金
     */
    private BigDecimal draftDocCommission;
    /**
     * 律师函多少钱
     */
    private BigDecimal lawyerLetterMoney;
    /**
     * 律师函佣金
     */
    private BigDecimal lawyerLetterCommission;

    /**
     * 律师各类服务实际收入金额
     *
     * @param type 类型
     *             1 电话咨询
     *             2 指定会面
     *             3 非指定会面
     *             4 修改文书
     *             5 草拟文书
     *             6 律师函
     * @return 返回金额
     */
    public BigDecimal buildMoney(int type) {
        switch (type) {
            case 1:
                return this.telConMoney.multiply(this.telConCommission);
            case 2:
                return this.meetMoney.multiply(this.meetCommission);
            case 3:
                return this.meetAllMoney.multiply(this.meetAllCommisson);
            case 4:
                return this.modifyDocMoney.multiply(this.modifyDocCommission);
            case 5:
                return this.draftDocMoney.multiply(this.draftDocCommission);
            case 6:
                return this.lawyerLetterMoney.multiply(this.lawyerLetterCommission);
            default:
                return new BigDecimal(0);
        }
    }
}

File: lecshop_api_modules/lecshop_web_admin/src/main/webapp/res/pagejs/profitset.js
// 基本路径
var basePath = $("#basePath").val();

// 初始化列表
var initData = function () {
    $.ajax({
        url: basePath + '/queryprofitset.htm',
        success: function (data) {
            if (data !== "" && data !== null) {
                $("#tab1 input[name='telConMoney']").val(data.telConMoney);
                $("#tab1 input[name='telConCommission']").val(decimalToInt(data.telConCommission));
                $("#tab2 input[name='meetAllMoney']").val(data.meetAllMoney);
                $("#tab2 input[name='meetAllCommisson']").val(decimalToInt(data.meetAllCommisson));
                $("#tab3 input[name='meetMoney']").val(data.meetMoney);
                $("#tab3 input[name='meetCommission']").val(decimalToInt(data.meetCommission));
                $("#tab4 input[name='withdrawTime']").val(data.withdrawTime);
                $("#tab5 input[name='modifyDocMoney']").val(data.modifyDocMoney);
                $("#tab5 input[name='modifyDocCommission']").val(decimalToInt(data.modifyDocCommission));
                $("#tab6 input[name='draftDocMoney']").val(data.draftDocMoney);
                $("#tab6 input[name='draftDocCommission']").val(decimalToInt(data.draftDocCommission));
                $("#tab7 input[name='lawyerLetterMoney']").val(data.lawyerLetterMoney);
                $("#tab7 input[name='lawyerLetterCommission']").val(decimalToInt(data.lawyerLetterCommission));
            }
        }
    });
};

// 初始化列表
initData();

// 刷新列表
var refreshData = function () {
    initData();
};

function saveBtn(num) {
    if (!$("#form" + num).valid()) {
        return;
    }
    var commission = $("#form" + num).find(".moneyRange").val();
    if (commission > 100 | commission < 0) {
        showerror("佣金百分比应在0至100之间");
        return;
    }
    var profitSet = {
        telConMoney: $("#tab1 input[name='telConMoney']").val(),
        telConCommission: intToDecimal($("#tab1 input[name='telConCommission']").val()),
        meetAllMoney: $("#tab2 input[name='meetAllMoney']").val(),
        meetAllCommisson: intToDecimal($("#tab2 input[name='meetAllCommisson']").val()),
        meetMoney: $("#tab3 input[name='meetMoney']").val(),
        meetCommission: intToDecimal($("#tab3 input[name='meetCommission']").val()),
        withdrawTime: $("#tab4 input[name='withdrawTime']").val(),
        modifyDocMoney: $("#tab5 input[name='modifyDocMoney']").val(),
        modifyDocCommission: intToDecimal($("#tab5 input[name='modifyDocCommission']").val()),
        draftDocMoney: $("#tab6 input[name='draftDocMoney']").val(),
        draftDocCommission: intToDecimal($("#tab6 input[name='draftDocCommission']").val()),
        lawyerLetterMoney: $("#tab7 input[name='lawyerLetterMoney']").val(),
        lawyerLetterCommission: intToDecimal($("#tab7 input[name='lawyerLetterCommission']").val())
    };
    LSFetch({
        url: basePath + '/editprofitset.htm?type=' + num,
        data: JSON.stringify(profitSet),
        contentType: "application/json;charset=utf-8",
        success: function (data) {
            if (data === 1) {
                $('#success_dialog').modal('show');
                refreshData();
            } else {
                showerror("编辑出错");
            }
        }
    });
}

//小数转整数
function decimalToInt(value) {
    return parseFloat(value) * 100
}

//整数转小数
function intToDecimal(value) {
    return parseFloat(value) / 100
}
